From 999a9121bcb26c451cd4ef9cdca6c36e0e96f6cd Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:00 -0400
Subject: [powerpc] Add more exception trampolines for hypervisor exceptions

Message-id: <1410545655-205645-72-git-send-email-dzickus@redhat.com>
Patchwork-id: 94092
O-Subject: [RHEL7 PATCH 071/626] powerpc: Add more exception trampolines for hypervisor exceptions
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit d671ddd6654daf93c56f6ddff827bed448e1a312
Author: Michael Ellerman <michael@ellerman.id.au>
Date:   Thu Aug 15 15:22:18 2013 +1000

    powerpc: Add more exception trampolines for hypervisor exceptions

    This makes back traces and profiles easier to read.

    Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/exceptions-64s.S b/arch/powerpc/kernel/exceptions-64s.S
index d805af6..3a9ed6a 100644
--- a/arch/powerpc/kernel/exceptions-64s.S
+++ b/arch/powerpc/kernel/exceptions-64s.S
@@ -293,27 +293,31 @@ system_call_pSeries:
   * out of line to handle them
   */
  . = 0xe00
-hv_exception_trampoline:
+hv_data_storage_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b h_data_storage_hv
 
  . = 0xe20
+hv_instr_storage_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b h_instr_storage_hv
 
  . = 0xe40
+emulation_assist_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b emulation_assist_hv
 
  . = 0xe60
+hv_exception_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b hmi_exception_hv
 
  . = 0xe80
+hv_doorbell_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b h_doorbell_hv
@@ -809,6 +813,7 @@ system_call_relon_pSeries:
  b . /* Can't happen, see v2.07 Book III-S section 6.5 */
 
  . = 0x4e40
+emulation_assist_relon_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b emulation_assist_relon_hv
@@ -817,6 +822,7 @@ system_call_relon_pSeries:
  b . /* Can't happen, see v2.07 Book III-S section 6.5 */
 
  . = 0x4e80
+h_doorbell_relon_trampoline:
  SET_SCRATCH0(r13)
  EXCEPTION_PROLOG_0(PACA_EXGEN)
  b h_doorbell_relon_hv
-- 
1.7.1